src/cls/rbd/cls_rbd_types.h

```diff
@@ -328,6 +336,68 @@ struct SnapshotNamespaceOnDisk {
 };
 WRITE_CLASS_ENCODER(SnapshotNamespaceOnDisk);
 
+SnapshotNamespaceType get_namespace_type(const SnapshotNamespace& snapshot_namespace);
+
+enum GroupSnapshotState {
+  GROUP_SNAPSHOT_STATE_PENDING = 0,
```

Comment:
Nit: can we rename this to ```GROUP_SNAPSHOT_STATE_INCOMPLETE```? If there is ever an issue, a PENDING group snapshot will never migrate to COMPLETE.